### ISSUE TITLE
Weighted Loss

### ISSUE THREAD

**Issue by geoffreyangus** [jrzech/reproduce-chexnet#8]
The original paper implements a weighted binary cross entropy loss. I don't see how you could have reproduced the results from the paper without it, as some tasks have nearly a 2:112 class imbalance. Am I missing something?

**Comment by jrzech** [comment]
I believe you are confusing the (admittedly confusingly named) CheXNet with the follow-up CheXNeXt. The CheXNet paper does not describe using a weighted loss function. The CheXNet 'Model Architecture and Training' section is limited and is given in its entirety below:

"CheXNet is a 121-layer Dense Convolutional Network (DenseNet) (Huang et al., 2016) trained on the ChestX-ray 14 dataset. DenseNets improve flow of information and gradients through the network, making the optimization of very deep networks tractable. We replace the final fully connected layer with one that has a single output, after which we apply a sigmoid nonlinearity.

The weights of the network are initialized with weights from a model pretrained on ImageNet (Deng et al., 2009). The network is trained end-to-end using Adam with standard parameters (β1 = 0.9 and β2 = 0.999) (Kingma & Ba, 2014). We train the model using mini- batches of size 16. We use an initial learning rate of 0.001 that is decayed by a factor of 10 each time the validation loss plateaus after an epoch, and pick the model with the lowest validation loss."

(https://arxiv.org/pdf/1711.05225.pdf)

There is no mention of a weighted loss function. In the follow-up CheXNeXt paper (https://www.ncbi.nlm.nih.gov/pubmed/30457988), however, a weighted loss function is described. 

The model trains and achieves AUCs similar to those reported in the original paper without a weighted loss function despite the large class imbalance, which exceeds 2:112 in some cases (e.g., hernia). Perhaps convergence is promoted because we are learning 14 findings simultaneously, and many cases are positive for at least one type of finding. While it may be surprising that the model converges without a weighted loss function, you can re-run the training code and verify for yourself that it does.

**Comment by jrzech** [comment]
It is somewhat surprising, but you'll find that the model converges and achieves AUCs comparable to those in the original paper despite the large class imbalance, which actually exceeds 2:112 in some cases (e.g., hernia). You can re-run the training code and verify this for yourself. Perhaps this is because we're learning 14 findings simultaneously, and many cases are positive for at least one finding.
